aggregator/src/test/java/org/jenkinsci/plugins/workflow/steps/stash/StashTest.java

```diff
@@ -69,4 +70,30 @@
         assertEquals("{}", StashManager.stashesOf(b).toString());
     }
 
+    @Issue("JENKINS-31086")
+    @Test public void testDefaultExcludes() throws Exception {
+        WorkflowJob p = r.jenkins.createProject(WorkflowJob.class, "p");
+        p.setDefinition(new CpsFlowDefinition(
+                "node {\n" +
+                        "  writeFile file: 'subdir/.gitignore', text: 'whatever'\n" +
+                        "  writeFile file: 'subdir/other', text: 'more'\n" +
+                        "  dir('subdir') {stash name:'whatever', useDefaultExcludes: false}\n" +
+                        "}\n" +
+                        "node {\n" +
+                        "  dir('elsewhere') {\n" +
+                        "    unstash 'whatever'\n" +
+                        "    echo \"got .gitignore: ${readFile '.gitignore'} other: ${readFile 'other'}\"\n" +
+                        "  }\n" +
+                        "  writeFile file: 'at-top', text: 'ignored'\n" +
+                        "  stash name: 'from-top', includes: 'elsewhere/'\n" +
+                        "  semaphore 'ending'\n" +
```

Comment:
Yeah, it is - the `unstash` is within `dir('elsewhere')`, and the original `stash` is done from within `subdir`, so we're not actually ever stashing the directory, just its contents.